_posts/2016-08-09-the-tech-behind-live-auction-integration.markdown

```diff
@@ -0,0 +1,69 @@
+
+Causality was developed in Scala, using the [Akka](http://doc.akka.io/docs/akka/current/intro/what-is-akka.html) technology suite for distributed computing. At its core is an append-only storage engine, based on [Akka Persistence](http://doc.akka.io/docs/akka/current/scala/persistence.html), with a small library we developed called [Atomic Store](https://github.com/artsy/atomic-store) that allowed us to achieve strict consistency, at the cost of maximal throughput -- a trade-off that is explored in the readme of that project.
+
+Lastly, Causality has an [Akka HTTP](http://doc.akka.io/docs/akka/current/scala/http/introduction.html)-based API layer, with a WebSocket server implemented using [Akka Streams](http://doc.akka.io/docs/akka/current/scala/stream/stream-introduction.html). Asynchronous updates generated in the event processing logic are published across the cluster using [Akka Distributed Pub/Sub](http://doc.akka.io/docs/akka/current/scala/distributed-pub-sub.html), and they are merged into the WebSocket outflow.
+
+# Pre-bidding, tooling, and other concerns
+
+In addition to accepting bids placed during a live sale, we also allow users to place bids before the event begins. In practice, this is almost the same workflow as our existing timed auction experience. For this reason, we chose to leverage all of our existing technology. The work of preparing our preexisting tech for LAI involved widespread modifications to our front-end UI, messaging services, admin tooling, and monitoring to make them appropriate for a live sale, as well as a reliable handoff of responsibility from these preexisting front- and back-end services to the new ones at the time the sale goes live.
+
+We relied on our automated test suites, as well as thorough manual testing by the entire Artsy auctions team, to ensure that this handoff functioned smoothly under various circumstances. We will eventually eliminate this duplication. But this will require delicate refactoring of our preexisting tooling, which we will take on, even as we execute a fall auction season significantly busier than the last.
+
+# Reflection
+
+In the process of architecting our LAI product, we had to make some tough decisions in the face of new challenges. Chief among these were the decisions on where on the spectrum of bleeding-edge technology versus tried-and-true choices to land, for many of our subcomponents. Bleeding-edge tech often offers more elegant and performant solutions, but at the cost of learning curve and risk of immaturity. We also had to carefully prioritize functionality. Choosing wisely througout the process was critical to shipping on time. The rationale behind these decisions and their outcomes will be the result of future pieces.
```

Comment:
This last sentence could just be "Future pieces will discuss the rationale behind these decisions and their outcomes."